File: src/main/java/br/com/teste/java/testebackend/domain/Parada.java
package br.com.teste.java.testebackend.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Builder
public class Parada implements Serializable {
    private static final long serialVersionUID = -3656431259068389491L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="parada_id", unique = true)
    private Long id;
    private String name;
    private double latitude;
    private double longitude;

    @ManyToMany(mappedBy = "paradas")
    private List<Linha> linhas = new ArrayList();

}

File: TransportePublico/TransportePublico.Domain/Parada.cs
using System.Collections.Generic;

namespace TransportePublico.Domain
{
    public class Parada
    {
        public Parada()
        {
            Linhas = new List<Linha>();
        }

        public long Id { get; set; }
        public string Name { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public List<Linha> Linhas { get; set; }
    }
}
